src/main/java/com/trueffect/logic/DateOperation.java

```diff
@@ -0,0 +1,155 @@
+            if (diff.getYears() > 0) {
+                return true;
+            } else if (diff.getYears() == 0) {
+                if (diff.getMonths() >= 0 && diff.getDays() >= 0) {
+                    return true;
+                }
+            }
+        } catch (ParseException ex) {
+        }
+        return false;
+    }
+
+    public static int getYearCurrent() {
+        Calendar cal = Calendar.getInstance();
+        return cal.get(Calendar.YEAR);
+    }
+
+    public static boolean areSameMonthAndYear(String dateFirst, String dateSecond) {
+        try {
+            SimpleDateFormat dateFormat = new SimpleDateFormat(ConstantData.SIMPLE_DATE_FORMAT);
+            Date dateFirstFormat = dateFormat.parse(dateFirst);
+            Date dateSecondFormat = dateFormat.parse(dateSecond);
+            LocalDate localDateFirst = LocalDate.fromDateFields(dateFirstFormat);
+            LocalDate localDateSecond = LocalDate.fromDateFields(dateSecondFormat);
+            Period diff = Period.fieldDifference(localDateFirst, localDateSecond);
+            if (diff.getYears() == 0 && diff.getMonths() == 0) {
+                return true;
+            }
+        } catch (ParseException ex) {
+        }
+        return false;
+    }
+
+    /*
+    *It is mandatory that dateSecond be equal to or greater than date dateFirst.
+     */
+    public static int diferenceMinutes(String dateFirst, String dateSecond) {
+        int diferenceMin = -1;
+        try {
+            SimpleDateFormat dateFormat = new SimpleDateFormat(ConstantData.TIMESTAMP_DATE_FORMAT);
+            Date dateFirstFormat = dateFormat.parse(dateFirst);
+            Date dateSecondFormat = dateFormat.parse(dateSecond);
+            LocalDate localDateFirst = LocalDate.fromDateFields(dateFirstFormat);
+            LocalDate localDateSecond = LocalDate.fromDateFields(dateSecondFormat);
+            Period diff = Period.fieldDifference(localDateFirst, localDateSecond);
+            diferenceMin = diff.getMinutes();
+        } catch (ParseException ex) {
+        }
+        return diferenceMin;
+    }
+      /*
+    *It is mandatory that dateSecond be equal to or greater than date dateFirst.
+     */
+    public static int diferenceDays(String dateFirst, String dateSecond) {
+        int diferenceDays = -1;
+        try {
+            SimpleDateFormat dateFormat = new SimpleDateFormat(ConstantData.TIMESTAMP_DATE_FORMAT);
+            Date dateFirstFormat = dateFormat.parse(dateFirst);
+            Date dateSecondFormat = dateFormat.parse(dateSecond);
+            LocalDate localDateFirst = LocalDate.fromDateFields(dateFirstFormat);
+            LocalDate localDateSecond = LocalDate.fromDateFields(dateSecondFormat);
+            Period diff = Period.fieldDifference(localDateFirst, localDateSecond);
+            diferenceDays = diff.getDays();
+        } catch (ParseException ex) {
+        }
+        return diferenceDays;
+    }
```

Comment:
EmployeeLogic

personLogic : variable not used
Either eitherInserted  : entiendo q el metodo devuelve una persona por que se utiliza  como variable  eitherInserted

eitherRes = jobCrud.getJobOfName : devuelve un job por que no colocar un nombre mas especifico que senale q es un objeto tipo job
eitherRes = employeeCrud.insertPhone : entiendo q inserta un phone  no deberia devolver un phone tambien
// Get the employees without the phones        eitherRes = employeeCrud.getEmployeeByIdentifier( 
: devuelve datos po numero telefonico actualizar la documentasion

public Either getEmployee(String typeId, String identifier) : Metodo no utilizado por favor eliminar metodo si no corresponde
getEmployee : no corresponde  colocar commit
update : mismas observaciones 
eitherRes = permission.getPerson
eitherRes = permission.checkUserPermission
eitherRes = operationModel.verifyId
eitherRes = getEmployee
eitherRes = jobCrud.getJobOf
eitherRes = personCrud.updatePerson
eitherRes = employeeCrud.getDataJob
employeeCrud.updateStatusDataJob
eitherRes = eitherRes = employeeCrud.insertDataJob
  eitherRes = getPhonesUpdate
   eitherRes = getPhonesInsert
Colocar nombres a las variables significativas
293 borrar comentarios que no corresponden 
getDataJobInsert   if (dataJobNew.getJobId() != 0) ** por que se hace la comparacion diferente a cero
bondAsigned usear un foreach  for (int i = 0; i < listDataJob.size(); i++) 
getIdBondOf : la asignacion de bonos debe de ser a fechas exactas.	

getSalaryOfAssignBond	: utilizar mapas para encontrar valores
getBondEmployee : utilizar mapas para encontrar valores
-MEJORAR LA LOGICA DEL METODO DE ASIGNACION DE BONOS